android/tangram/src/com/mapzen/tangram/MapData.java

```diff
@@ -0,0 +1,102 @@
+package com.mapzen.tangram;
+
+import java.util.List;
+
+public class MapData {
+
+    /**
+     * Construct a new MapData object for adding drawable data to the map
+     * @param name Name of the data source in the scene file for styling this object's data
+     */
+    public MapData(String name) {
+        this.name = name;
+        this.id = addDataSource(name);
+    }
+
+    /**
+     * Get the name of this data source
+     * @return The name
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Remove all data from this source
+     * @return This object, for chaining
+     */
+    public MapData clear() {
+        clearSourceData(id);
+        return this;
+    }
+
+    /**
+     * Add geometry from a GeoJSON string to this data source
+     * @param data String of GeoJSON containing a Feature or FeatureCollection
+     * @return This object, for chaining
+     */
+    public MapData addGeoJSON(String data) {
+        addSourceGeoJSON(id, data);
+        return this;
+    }
+
+    /**
+     * Add a point geometry to this data source
+     * @param point LngLat with the coordinates of the point
+     * @return This object, for chaining
+     */
+    public MapData addPoint(LngLat point) {
```

Comment:
With regards to search results we will only need the index of the selected feature in the initial version. If the order of points added is preserved this is enough.

If the order is not preserved we could include the index (or a feature id) as an additional argument.